src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/OldAndBrokenServiceLocatorAnalyzer.cs

```diff
@@ -30,105 +42,104 @@ bool excludeKnownProblemDlls
 		}
 
 		public void RegisterServiceLocatorAnalyzer( CompilationStartAnalysisContext context ) {
-			// Cache some important type lookups
-			var locatorType = context.Compilation.GetTypeByMetadataName( "D2L.LP.Extensibility.Activation.Domain.OldAndBrokenServiceLocator" );
-			var factoryType = context.Compilation.GetTypeByMetadataName( "D2L.LP.Extensibility.Activation.Domain.OldAndBrokenServiceLocatorFactory" );
-			var activatorType = context.Compilation.GetTypeByMetadataName( "D2L.LP.Extensibility.Activation.Domain.IObjectActivator" );
-			var customActivatorType = context.Compilation.GetTypeByMetadataName( "D2L.LP.Extensibility.Activation.Domain.ICustomObjectActivator" );
-			var staticDILocatorType = context.Compilation.GetTypeByMetadataName( "D2L.LP.Extensibility.Activation.Domain.Default.StaticDI.StaticDILocator" );
-
-			// If those type lookups failed then OldAndBrokenServiceLocator
-			// cannot resolve and we don't need to register our analyzer.
 
-			if( locatorType == null || locatorType.Kind == SymbolKind.ErrorType ) {
-				return;
-			}
-			if ( factoryType == null || factoryType.Kind == SymbolKind.ErrorType ) {
+			ImmutableHashSet<ITypeSymbol> disallowedTypes = GetDisallowedTypes( context.Compilation );
+			if( disallowedTypes.IsEmpty ) {
 				return;
 			}
 
-			ImmutableArray<INamedTypeSymbol> disallowedTypes = ImmutableArray.Create(
-				locatorType,
-				factoryType,
-				activatorType,
-				customActivatorType,
-				staticDILocatorType
-			);
-
 			AllowedTypeList allowedTypeList = AllowedTypeList.CreateFromAnalyzerOptions(
 				allowedListFileName: "OldAndBrokenServiceLocatorAllowedList.txt",
 				analyzerOptions: context.Options
 			);
 
-			//Prevent static usage of OldAndBrokenServiceLocator
-			//For example, OldAndBrokenServiceLocator.Instance.Get<IFoo>()
-			context.RegisterSyntaxNodeAction(
-				ctx => PreventOldAndBrokenUsage(
-					ctx,
-					disallowedTypes,
-					allowedTypeList
-				),
-				SyntaxKind.IdentifierName
+			TypeRuleSets typeRules = new( allowedTypeList, disallowedTypes );
+

```

Comment:
The 5 disallowed types only have methods & properties, so registering for those operations